src/main/java/spring/bookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,36 @@
+package spring.bookstore.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import spring.bookstore.dto.UserRegistrationRequestDto;
+import spring.bookstore.dto.UserResponseDto;
+import spring.bookstore.exception.RegistrationException;
+import spring.bookstore.mapper.UserMapper;
+import spring.bookstore.model.User;
+import spring.bookstore.repository.UserRepository;
+
+@RestController
+@RequestMapping("/api/auth")
```

Comment:
add `server.servlet.context-path=/api` to application.properties

```suggestion
@RequestMapping("/auth")
```